src/azure-cli/azure/cli/command_modules/appservice/static_sites.py

```diff
@@ -577,10 +584,11 @@ def link_user_function(cmd, name, resource_group_name, function_resource_id, for
     function = StaticSiteUserProvidedFunctionAppARMResource(function_app_resource_id=function_resource_id,
                                                             function_app_region=function_location)
 
-    return client.begin_register_user_provided_function_app_with_static_site(
+    return client.begin_register_user_provided_function_app_with_static_site_build(
         name=name,
         resource_group_name=resource_group_name,
         function_app_name=function_name,
+        environment_name=environment_name,
         static_site_user_provided_function_envelope=function,
         is_forced=force)
 
```

Comment:
I worry that this might be a breaking change to existing users the way this is currently implemented. Based on the API response when I test this, it seems like the type of azure resource created is different (new type is `Microsoft.Web/staticSites/builds/userProvidedFunctionApps` whereas old type is `Microsoft.Web/staticSites/userProvidedFunctionApps`) . Presumably these would behave differently since one is scoped for the whole app and one is scoped to a single environment. 

I would suggest that we remove the default `"default"` for the parameter `environment_name` and only call the new SDK method if the `environment_name` is not `None` (and otherwise call the old method without environment name).

Reply:
Makes sense. I've now implemented your suggestion. Hope this is what you had in mind.